### PR TITLE
lantiq-xway:  support for AVM FRITZ!Box 7312

### PR DESCRIPTION
**Pull request by misanthropos**
- [x] must be flashable from vendor firmware
  - [ ] webinterface
  - [ ] tftp
  - [x] other: EVA
- [x] must support upgrade mechanism
  - [x] must have working sysupgrade
    - [x] must keep/forget configuration (if applicable)
      *think `sysupgrade [-n]` or `firstboot`*
  - [?] must have working autoupdate
    *usually means: gluon profile name must match image name*
- [x] xxxxx/xxx/phone button must return device into config mode
- [ ] primary mac should match address on device label (or packaging) (https://gluon.readthedocs.io/en/latest/dev/hardware.html#notes)
- wired network
  - [x] should support all network ports on the device
  - [x] must have correct port assignment (WAN/LAN)
- wifi (if applicable)
  - [x] association with AP must be possible on all radios
  - [x] association with 802.11s mesh must be working on all radios 
  - [x] ap/mesh mode must work in parallel on all radios
- led mapping
  - power/sys led (_critical, because led definitions are setup on firstboot only_)
    - [x] lit while the device is on
    - [x] should display config mode blink sequence 
(https://gluon.readthedocs.io/en/latest/features/configmode.html)
  - radio leds
    - [x] should map to their respective radio
    - [x] should show activity
  - switchport leds
    - [ ] should map to their respective port (or switch, if only one led present) 
    - [ ] should show link state and activity
- outdoor devices only
  - [ ] added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`
~

### CODE REVIEW

**Comment by blocktrron** [comment]
Please fix the typo in the commit message.

File: docs/user/supported_devices.rst [hunk 1]
```diff
@@ -21,6 +21,7 @@ ar71xx-generic
 * AVM
 
   - Fritz!Box 4020 [#avmflash]_
+  - Fritz!Box 7312 [#avmflash]_
```
Comment (blocktrron) [inline docs/user/supported_devices.rst]: This is not the correct place, as the 7312 is not part of the ar71xx-generic target. Add a new section for the lantiq-xway target and add the FRITZ!Box 7312 there.

File: docs/user/supported_devices.rst [hunk 2]
```diff
@@ -224,6 +224,15 @@ ipq806x-generic [#80211s]_
 
   - R7800
 
+
+lantiq-xway
+-----------
+
+* AVM
+
+  - Fritz!Box 7312 [#avmflash]_
```
Comment (mweinelt) [inline docs/user/supported_devices.rst]:
```suggestion
  - FRITZ!Box 7312 [#avmflash]_
```

Comment (blocktrron) [reply docs/user/supported_devices.rst]: Actually - we made this mistake at some places already - just like the inconsistent whitespacing. I would merge this as it is and clean up both points after - Is this okay with you?

Comment (mweinelt) [reply docs/user/supported_devices.rst]: Fine with me.

**Comment by blocktrron** [comment]
I've made some changes to your commit: 

 - Remove the unnecessary whitespace newlines in the docs
 - added an `add` before `support for ...` to your title

Merging this now.

**Comment by ecsv** [comment]
Which button is used here for the config mode? Because a user just complained that neither the dect nor the wlan button will do anything for him:

https://github.com/openwrt/openwrt/blob/e3eaf578082ee26ebc0021faed8ae42aca007f9e/target/linux/lantiq/files-5.4/arch/mips/boot/dts/lantiq/ar9_avm_fritz7312.dts#L30

**Comment by misanthropos** [comment]
Assuming it runs gluon (meaning not OpenWRT) you press the DECT button
for 3 seconds.

See: https://gluon.readthedocs.io/en/latest/features/configmode.html

I believe the vanilla OpenWRT does not offer a config mode key.

**Comment by ecsv** [comment]
Thanks for confirming this.

For some reason, it doesn't seem to work for him. But I don't have access to the device and trying remotely to assist him in getting the device up and running. And now it seems like he isn't happy about my word "complained". So I will stop for now.

**Pull request merged**